Lib/test/test_capi/test_misc.py

```diff
@@ -1626,6 +1627,41 @@ def test_tp_mro_is_set(self):
                 self.assertIsNot(mro, None)
 
 
+class TestStaticTypes(unittest.TestCase):
+
+    _has_run = False
+
+    @classmethod
+    def setUpClass(cls):
+        # The tests here don't play nice with our approach to refleak
+        # detection, so we bail out in that case.
+        if cls._has_run:
+            raise unittest.SkipTest('these tests do not support re-running')
+        cls._has_run = True
```

Comment:
This is nice. IMO, we should use a similar approach in `test_import` instead of the hack suggested in gh-104796.

Reply:
Perhaps also gh-105085 should have been solved like this.

Reply:
I'm glad you appreciate the solution 😄, but actually am a bit torn about this.  On the one hand, it gets us what we want in a simple, clear way (IMHO).  On the other hand, this approach means there *may* be code paths that don't get checked for ref leaks, which we'd want to avoid.

If we use this pattern elsewhere, we'll need to be vigilant about making sure we don't miss any ref leak coverage.  I'm not sure how easy that would be generally, though it's unlikely that there are more than a handful of tests where we'd want to skip refleak detection.  FWIW, in this case I don't think ref leaks are much of a concern.

Reply:
> we should use a similar approach

Wouldn't this skip the whole `class Test` instead of a smaller `def test_...`?

Reply:
Yes, so you make sure to collect all your "ref leak unsafe" tests in a single unittest class.

Reply:
Alternatively, create a decorator for it.